src/app_service.rs

```diff
@@ -245,7 +245,7 @@ where
             inner.path.reset();
             inner.head = head;
             inner.payload = payload;
-            inner.app_data = self.data.clone();
+            inner.app_data.push(self.data.clone());
```

Comment:
I'm not 100% clear on the purpose and lifecycle of the request pool so not sure if this should be `.app_data.push(...)` or `.app_data = tiny_vec![...]`.

Reply:
I think the current is fine.